include/zephyr/arch/common/ffs.h

```diff
@@ -37,6 +37,30 @@ static ALWAYS_INLINE unsigned int find_msb_set(uint32_t op)
 	return 32 - __builtin_clz(op);
 }
 
+/**
+ *
+ * @brief find most significant bit set in a 64-bit word
+ *
+ * This routine finds the first bit set starting from the most significant bit
+ * in the argument passed in and returns the index of that bit.  Bits are
+ * numbered starting at 1 from the least significant bit.  A return value of
+ * zero indicates that the value passed is zero.
+ *
+ * @return most significant bit set, 0 if @a op is 0
+ */
+
+static ALWAYS_INLINE unsigned int find_msb_set_64(uint64_t op)
+{
+	if (op == 0) {
```

Comment:
There already exists `u64_count_leading_zeros` and `u64_count_trailing_zeros` (and 32-bit variants for these). Why do we need these functions, or can we call those functions here in order to avoid duplicate implementations?

Reply:
Good point - I think we likely don't even need `find_msb_set_32()`.

Reply:
Fixed!